Prevents Regal Rats from being possessed pre-round

Rhials opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

You can no longer observe and assume control of a regal rat during pre-round.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Closes #70018.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: You can no longer assume control of Regal Rats during preround.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rhials committed:
rat fix rat fix

Mothblocks reviewed (approved):
In a perfect world I think this would be a component or similar, I don't like that this can easily just happen again when someone adds a new mob.

Mothblocks commented:
Human fly trap monsters have it too, though I don't know if they re-implement it themselves or are just subtyping off of something

Rhials committed:
draft one

Rhials committed:
for clarification

Mothblocks commented on code/__DEFINES/dcs/signals/signals_mind.dm:
```diff
@@ -6,3 +6,6 @@
 #define COMSIG_ANTAGONIST_REMOVED "antagonist_removed"
+
+/// Called when an observer attempts to possess a ghost_role_spawnpoint component-bearing mob
+#define COMSIG_ATTEMPT_POSSESSION "attempt_possession" //Find a better file to put this in
```
Just make a `signals_ghost_role_spawnpoint.dm` file

Mothblocks commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -132,21 +132,20 @@
 	if(become_gorilla != "Yes" || QDELETED(src) || QDELETED(user) || being_polled_for || mind || client)
 		return
 
+	SEND_SIGNAL(src, COMSIG_ATTEMPT_POSSESSION, user)
```
The entire point is that the entire proc this is in shouldn't exist and should be handled by the component

Mothblocks commented on code/modules/mob/living/simple_animal/hostile/regalrat.dm:
```diff
@@ -74,17 +75,11 @@
-	key = user.key
-	src.log_message("took control of [name].", LOG_GAME)
+
+	SEND_SIGNAL(src, COMSIG_ATTEMPT_POSSESSION, user)
```
The entire point is that the entire proc this is in shouldn't exist and should be handled by the component

Mothblocks commented on code/datums/components/ghost_role_spawnpoint.dm:
```diff
@@ -0,0 +1,36 @@
+/datum/component/ghost_role_spawnpoint //Small component, but it helps to consolidate general observer role spawning behavior/checks into one place.
```
```suggestion
/// Write useful documentation here for what this does and how to use it
/datum/component/ghost_role_spawnpoint
```

Mothblocks reviewed (changes requested):
Well if we're going to do the best path then code review is about to get more serious.

You can't have the component be like this. The component should handle everything on its own, and the only thing consumers should have to do is add it, maybe with the text they want to see.

Rhials committed:
progress

moves all regal rat possession code (except for the ghost poll) to the component

Ask text is now handled in the component setup

COMSIG_ATOM_ATTACK_GHOST now triggers possession instead of a new signal

Changes the cargorilla post-possession messages and role assignments to a new proc, linked to the mind transfer signal

Rhials committed:
Revert "progress"

This reverts commit 070c7708f5291379d2cdb14ae3d6a6b1c241d990.

Rhials committed:
Revert "for clarification"

This reverts commit f8463dc4ebf78b450d4b7052846ee4399e38d62c.

Rhials committed:
Revert "draft one"

This reverts commit b8cfdb8d16172d42cc6835359cce94cca7bed38d.

Rhials commented:
Alright, I figure that I'm just out of my depth on this one. I intend to return and make this a component when I am smarter, but there's not point in holding up the fix until then.

Rhials committed:
oh, and this

Mothblocks commented:
Fair enough
